src/main/java/com/snowflake/kafka/connector/internal/IngestSdkProvider.java

```diff
@@ -0,0 +1,102 @@
+
+  private LoggerHandler LOGGER;
+  private int streamingIngestClientCount;
+  private SnowflakeStreamingIngestClient streamingIngestClient;
+
+  /**
+   * This is a wrapper for the ingest sdk to help define a contract between KC and Ingest
+   *
+   * <p>Ideally all ingest sdk calls will go through this to make mocking and integration tests
+   * easier to write (potentially also to run most of the integration tests without needing a
+   * snowflake connection). Currently it only manages the client.
+   */
+  public IngestSdkProvider() {
+    LOGGER = new LoggerHandler(this.getClass().getName());
+    this.streamingIngestClientCount = 0;
+  }
+
+  public SnowflakeStreamingIngestClient createStreamingClient(
+      Map<String, String> connectorConfig, String connectorName) {
+    Map<String, String> streamingPropertiesMap =
+        StreamingUtils.convertConfigForStreamingClient(new HashMap<>(connectorConfig));
+    Properties streamingClientProps = new Properties();
+    streamingClientProps.putAll(streamingPropertiesMap);
+
+    String streamingIngestClientName = this.getStreamingIngestClientName(connectorName);
+
+    try {
+      LOGGER.info("Creating Streaming Client. ClientName:{}", streamingIngestClientName);
+      this.streamingIngestClientCount++;
+      this.streamingIngestClient =
+          SnowflakeStreamingIngestClientFactory.builder(streamingIngestClientName)
+              .setProperties(streamingClientProps)
+              .build();
+      return this.getStreamingIngestClient();
+    } catch (SFException ex) {
+      // note: unable to test this exception since the factory is not mockable. be careful changing
+      // logic here
+      LOGGER.error(
+          "Exception creating streamingIngestClient with name:{}", streamingIngestClientName);
+      throw new ConnectException(ex);
+    }
+  }
+
+  public void closeStreamingClient() {
+    String streamingIngestClientName = this.streamingIngestClient.getName();
+    LOGGER.info("Closing Streaming Client:{}", streamingIngestClientName);
+    try {
+      this.streamingIngestClient.close();
+    } catch (Exception e) {
+      // note: unable to test this exception since the factory is not mockable. be careful changing
+      // logic here
+      LOGGER.error(
+          "Failure closing Streaming client msg:{}, cause:{}",
+          e.getMessage(),
+          Arrays.toString(e.getCause().getStackTrace()));
+    }
+  }
+
+  public SnowflakeStreamingIngestClient getStreamingIngestClient() {
+    if (this.streamingIngestClient != null && !this.streamingIngestClient.isClosed()) {
+      return this.streamingIngestClient;
+    }
+
+    LOGGER.error("Streaming ingest client was null or closed. It must be initialized");
+    throw SnowflakeErrors.ERROR_3009.getException();
+  }
+
+  private String getStreamingIngestClientName(String connectorName) {
+    return STREAMING_CLIENT_PREFIX_NAME + connectorName + this.streamingIngestClientCount;
```

Comment:
Umm, not sure if this is right way to name the client. 
TaskIds would have been unique on all instances IMO, streamingIngestClientCount would start from 0 on every node. 

This should be unique IMO. [Unless it is okay to have same client names]

Reply:
something like `getKcInstanceId`

Reply:
Changed, thankss